onnxruntime/test/onnx/runner.cc

```diff
@@ -49,13 +49,32 @@ void ORT_CALLBACK RunTestCase(ORT_CALLBACK_INSTANCE pci, void* context, ORT_WORK
   }
 }
 
-void PTestRunner::Start(ORT_CALLBACK_INSTANCE, size_t concurrent_runs) {
-  concurrent_runs = std::min<size_t>(std::max<size_t>(1, concurrent_runs), c_.GetDataCount());
-  next_test_to_run = 0;
-  for (size_t i = 0; i != concurrent_runs; ++i) {
-    if (!ScheduleNew()) {
-      throw std::runtime_error("ScheduleNew task failed");
+void PTestRunner::Start(ORT_CALLBACK_INSTANCE pci, size_t concurrent_runs) noexcept {
+  bool atleast_one_run_scheduled = false;
+  try {
+    concurrent_runs = std::min<size_t>(std::max<size_t>(1, concurrent_runs), c_.GetDataCount());
+    next_test_to_run = 0;
+    for (size_t i = 0; i != concurrent_runs; ++i) {
+      if (!ScheduleNew()) {
+        if (i == 0) {
+          // This should be a rare case.
+          // Log it. While existing Finish will be called which will take care of the cleanup
+          LOGF_DEFAULT(ERROR, "Could not schedule tasks for test case: %s\n", c_.GetTestCaseName().c_str());
```

Comment:
Should we call out that we couldn't schedule the first run?
Also, we don't want to log for i>0 if ScheduleNew() returned false?

Reply:
I think we should always log

---
In reply to: [452672254](https://github.com/microsoft/onnxruntime/pull/4475#discussion_r452672254) [](ancestors = 452672254)

Reply:
I did not add a log statement when i > 0 and ScheduleNew returned false because that will almost in all the cases will be when there was no task to schedule and this is OK. I can log a warning message in this case instead of error.

Reply:
I take it back

---
In reply to: [452918421](https://github.com/microsoft/onnxruntime/pull/4475#discussion_r452918421) [](ancestors = 452918421,452672254)